client/src/features/modal/Modal.jsx

```diff
@@ -32,7 +32,12 @@ const Modal = ({ children, context }) => {
         onExitComplete={() => null}
       >
         {context.isOpen && (
-          <Backdrop onClick={context.handleClose}>
+          <Backdrop
+            onClick={() => {
+              context.handleClose();
+              if (typeof onBackdropClick === "function") onBackdropClick();
```

Comment:
I don't think this conditional should silently pass if a "function" typed `onBackdropClick` isn't passed as a prop.

If someone is adding more modals later but accidentally passes an `onBackdropClick` of some type _other than_ "function", this `if` will essentially hide that error.

Instead, it might be better to just check for the existence of `onBackdropClick`. Then if a non-function prop is passed to `Modal` it will easily be caught during testing.

```suggestion
              if (onBackdropClick) onBackdropClick();
```